content/blogposts/2024/anatomy-of-a-coding-assistant.md

```diff
@@ -12,25 +12,25 @@ heroImage: https://storage.googleapis.com/sourcegraph-assets/blog/anatomy/1_poin
 socialImage: https://storage.googleapis.com/sourcegraph-assets/blog/anatomy/1_pointer.png
 ---
 
-![1_pointer](https://storage.googleapis.com/sourcegraph-assets/blog/anatomy/1_pointer.png)
+![Pointer clicking on code snippet](https://storage.googleapis.com/sourcegraph-assets/blog/anatomy/1_pointer.png)
 
 Have you ever wondered how an AI coding assistant works?
 
 We’ve been working on our AI coding assistant, [Cody](https://sourcegraph.com/cody), for a while now, and over time we’ve learned that really, ***context is king***.
 
-![2_octopus](https://storage.googleapis.com/sourcegraph-assets/blog/anatomy/2_octopus.png)
+![Octopus with a crown labeled 'Context' and code snippets](https://storage.googleapis.com/sourcegraph-assets/blog/anatomy/2_octopus.png)
 
 If you ask ChatGPT a coding question about your particular codebase, it won’t be able to answer it because it doesn’t know anything about your codebase.
 
-![3_chatgpt](https://storage.googleapis.com/sourcegraph-assets/blog/anatomy/3_chatgpt.png)
+![ChatGPT interface showing a conversation on converting text into embeddings in Sourcegraph's Cody. The response explains three steps: 1. Choose a language model capable of generating embeddings, such as OpenAI's GPT or Google's BERT. 2. Prepare your environment by installing necessary libraries like `transformers` or `sentence-transformers`. 3. Generate embeddings, with an example command to install the `sentence-transformers` library using pip: `pip install sentence-transformers`. The response is displayed in a dark theme interface with a copy code button for the installation command.](https://storage.googleapis.com/sourcegraph-assets/blog/anatomy/3_chatgpt.png)
```

Comment:
These alts are way too long according to https://css-tricks.com/just-how-long-should-alt-text-be/ (unless I'm out of date)

Those should be in an  `aria-description` which is hard using Markdown syntax. Your call just my observation. =)